### PR TITLE
Add syspurpose entries to facts

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
This should parse out the `syspurpose.json` if defined and add those entries to the facts reported.

#### This Pull Request (PR) fixes the following issues
Fixes: #120

### CODE REVIEW

File: lib/facter/rhsm.rb [hunk 1]
```diff
@@ -31,6 +32,11 @@
     end
   end
 
+  # Find any syspurpose defined
+  chunk(:syspurpose) do
+    Puppet::Util::Json.load_file_if_valid('/etc/rhsm/syspurpose/syspurpose.json') if File.exist? '/etc/rhsm/syspurpose/syspurpose.json'
```
Comment (bastelfreak) [inline lib/facter/rhsm.rb:37]: I checked some servers and they all have the file with content `{}`. do you want to use the file in that case and basically create an empty fact?

Comment (jcpunk) [reply lib/facter/rhsm.rb:37]: I'm not sure what feels more natural for querying the settings....

On the one hand, wraping everything in `if X in Y` twice (if file has content, if setting is set) is a bit ugly.  On the other, a fact whose value is `{}` is really meaningless data.

Comment (jcpunk) [reply lib/facter/rhsm.rb]: After thinking on it a bit, folks storing the data in puppetdb would probably not have the fact set to `{}`

**Review by bastelfreak — APPROVED** [review]
looks good to me. @vchepkov could you take a look as well?

**Comment by vchepkov** [comment]
I am a big proponent of using module namespace to avoid clashing, if somebody already created similar fact.
I am not sure if it is set as a standard somewhere, but this notation seems to be reasonable
https://github.com/voxpupuli/puppet-extlib/blob/master/lib/facter/extlib__puppet_config.rb#L3

**Comment by jcpunk** [comment]
Since I'm looking to extend the existing fact, I'm not sure switching to a namespaced version is the right way to go...

**Pull request merged**